%% Generated by Sphinx.
\def\sphinxdocclass{report}
\documentclass[letterpaper,10pt,english]{sphinxmanual}
\ifdefined\pdfpxdimen
   \let\sphinxpxdimen\pdfpxdimen\else\newdimen\sphinxpxdimen
\fi \sphinxpxdimen=.75bp\relax
\ifdefined\pdfimageresolution
    \pdfimageresolution= \numexpr \dimexpr1in\relax/\sphinxpxdimen\relax
\fi
%% let collapsible pdf bookmarks panel have high depth per default
\PassOptionsToPackage{bookmarksdepth=5}{hyperref}

\PassOptionsToPackage{warn}{textcomp}
\usepackage[utf8]{inputenc}
\ifdefined\DeclareUnicodeCharacter
% support both utf8 and utf8x syntaxes
  \ifdefined\DeclareUnicodeCharacterAsOptional
    \def\sphinxDUC#1{\DeclareUnicodeCharacter{"#1}}
  \else
    \let\sphinxDUC\DeclareUnicodeCharacter
  \fi
  \sphinxDUC{00A0}{\nobreakspace}
  \sphinxDUC{2500}{\sphinxunichar{2500}}
  \sphinxDUC{2502}{\sphinxunichar{2502}}
  \sphinxDUC{2514}{\sphinxunichar{2514}}
  \sphinxDUC{251C}{\sphinxunichar{251C}}
  \sphinxDUC{2572}{\textbackslash}
\fi
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{amsmath,amssymb,amstext}
\usepackage{babel}



\usepackage{tgtermes}
\usepackage{tgheros}
\renewcommand{\ttdefault}{txtt}



\usepackage[Bjarne]{fncychap}
\usepackage{sphinx}

\fvset{fontsize=auto}
\usepackage{geometry}


% Include hyperref last.
\usepackage{hyperref}
% Fix anchor placement for figures with captions.
\usepackage{hypcap}% it must be loaded after hyperref.
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}

\addto\captionsenglish{\renewcommand{\contentsname}{Contents:}}

\usepackage{sphinxmessages}
\setcounter{tocdepth}{1}



\title{TissUUmaps}
\date{Apr 12, 2022}
\release{3.0}
\author{Nicolas Pielawski\and Axel Andersson\and Christophe Avenel\and Andrea Behanova\and Eduard Chelebian\and Anna Klemm\and Fredrik Nysjö\and Leslie Solorzano\and Carolina Wählby}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{Release}
\makeindex
\begin{document}

\pagestyle{empty}
\sphinxmaketitle
\pagestyle{plain}
\sphinxtableofcontents
\pagestyle{normal}
\phantomsection\label{\detokenize{index::doc}}


\sphinxAtStartPar
This page hosts the documentation for TissUUmaps 3.0.

\sphinxAtStartPar
For more information on the TissUUmaps project, including video tutorials and demos, visit our website: \sphinxurl{https://tissuumaps.github.io}.

\begin{sphinxShadowBox}
\sphinxstylesidebartitle{Work in progress!}

\sphinxAtStartPar
This page is mostly empty for now. We are working actively on writing this documentation, more content will be available soon!
\end{sphinxShadowBox}

\sphinxstepscope


\chapter{Introduction}
\label{\detokenize{docs/intro/index:introduction}}\label{\detokenize{docs/intro/index::doc}}
\sphinxstepscope


\section{About TissUUmaps}
\label{\detokenize{docs/intro/about:about-tissuumaps}}\label{\detokenize{docs/intro/about::doc}}
\sphinxAtStartPar
\sphinxhref{https://tissuumaps.github.io/}{TissUUmaps} is a browser\sphinxhyphen{}based tool for GPU\sphinxhyphen{}accelerated visualization and interactive exploration of 10\textasciicircum{}7+ datapoints overlaying tissue samples. Users can visualize markers and regions, explore spatial statistics and quantitative analyses of tissue morphology, and assess the quality of decoding in situ transcriptomics data. TissUUmaps provides instant multi\sphinxhyphen{}resolution image viewing, can be customized, shared, and also integrated in Jupyter Notebooks. We envision TissUUmaps to contribute to broader dissemination and flexible sharing of large\sphinxhyphen{}scale spatial omics data.

\sphinxAtStartPar
Currently, microscopy data can be cumbersome to share: physically transferring the images is often necessary and dedicated software must be installed. Instead, researchers can now share their findings with a simple link to a website running TissUUmaps. The images are loaded in real time, together with annotations, markers, and masks that may also be modified by the user. We also provide tools for quality control and image processing. The software is designed to display and interact with images at multiple resolutions and large numbers of markers, especially data from spatially resolved omics techniques and tissue atlases. TissUUmaps is compatible with many different bioimage informatics tools, and provides new ways to develop insights when exploring and sharing data.

\sphinxAtStartPar
You can access the \sphinxhref{https://tissuumaps.github.io/gallery/}{TissUUmaps project gallery} with interactive examples to explore data from in situ sequencing and spatial transcriptomics experiments and view localized quantification of cell and tissue morphology, including links to publications. For seeing examples of TissUUmaps compatibility with other platforms you can access the \sphinxhref{https://tissuumaps.github.io/tutorials/}{tutorials page}.

\sphinxstepscope


\section{Installation}
\label{\detokenize{docs/intro/installation:installation}}\label{\detokenize{docs/intro/installation::doc}}
\sphinxAtStartPar
\sphinxhref{https://tissuumaps.github.io/}{TissUUmaps} is a browser\sphinxhyphen{}based tool for fast visualization and exploration of millions of data points overlaying a tissue sample. TissUUmaps can be used as a web service or locally in your computer, and allows users to share regions of interest and local statistics.


\subsection{Windows installation}
\label{\detokenize{docs/intro/installation:windows-installation}}\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Download the Windows Installer from \sphinxhref{https://github.com/TissUUmaps/TissUUmaps/releases/latest}{the last release} and install it. Note that the installer is not signed yet and may trigger warnings from the browser and from the firewall. You can safely pass these warnings.

\end{enumerate}


\subsection{PIP installation (for Linux and Mac)}
\label{\detokenize{docs/intro/installation:pip-installation-for-linux-and-mac}}\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Install \sphinxcode{\sphinxupquote{libvips}} for your system: \sphinxurl{https://www.libvips.org/install.html}

\sphinxAtStartPar
An easy way to install \sphinxcode{\sphinxupquote{libvips}} is to use an \sphinxhref{https://docs.anaconda.com/anaconda/install/index.html}{Anaconda} environment with \sphinxcode{\sphinxupquote{libvips}}:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
conda create \PYGZhy{}y \PYGZhy{}n tissuumaps\PYGZus{}env \PYGZhy{}c conda\PYGZhy{}forge \PYG{n+nv}{python}\PYG{o}{=}\PYG{l+m}{3}.9 libvips
conda activate tissuumaps\PYGZus{}env
\end{sphinxVerbatim}

\item {} 
\sphinxAtStartPar
Install the TissUUmaps library using \sphinxcode{\sphinxupquote{pip}}:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
pip install \PYG{l+s+s2}{\PYGZdq{}TissUUmaps[full]\PYGZdq{}}
\end{sphinxVerbatim}

\item {} 
\sphinxAtStartPar
Start the TissUUmaps user interface:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
tissuumaps
\end{sphinxVerbatim}

\item {} 
\sphinxAtStartPar
Or start TissUUmaps as a local server:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
tissuumaps\PYGZus{}server path\PYGZus{}to\PYGZus{}your\PYGZus{}images
\end{sphinxVerbatim}

\sphinxAtStartPar
And open \sphinxurl{http://127.0.0.1:5000/} in your favorite browser.

\end{enumerate}

\sphinxstepscope


\section{Citing TissUUmaps}
\label{\detokenize{docs/intro/citing:citing-tissuumaps}}\label{\detokenize{docs/intro/citing::doc}}
\sphinxAtStartPar
Please cite our \sphinxhref{https://www.biorxiv.org/content/10.1101/2022.01.28.478131v1}{preprint} on bioRxiv if using TissUUmaps in your work:

\sphinxAtStartPar
\sphinxstylestrong{TissUUmaps 3: Interactive visualization and quality assessment of large\sphinxhyphen{}scale spatial omics data}\sphinxstyleemphasis{Nicolas Pielawski, Axel Andersson, Christophe Avenel, Andrea Behanova, Eduard Chelebian, Anna Klemm, Fredrik Nysjö, Leslie Solorzano, Carolina Wählby}bioRxiv 2022.01.28.478131; doi: https://doi.org/10.1101/2022.01.28.47813

\sphinxstepscope


\section{Changelog}
\label{\detokenize{docs/intro/versions:changelog}}\label{\detokenize{docs/intro/versions::doc}}

\subsection{3.0.8.5}
\label{\detokenize{docs/intro/versions:id1}}\begin{itemize}
\item {} 
\sphinxAtStartPar
Minor fixes.

\end{itemize}


\subsection{3.0.8.4}
\label{\detokenize{docs/intro/versions:id2}}\begin{itemize}
\item {} 
\sphinxAtStartPar
Add tiling to viewport capture for higher resolution output

\item {} 
\sphinxAtStartPar
Increase resolution of markers on high resolution devices

\item {} 
\sphinxAtStartPar
Fix jumps on pan with mouse gesture (mobile)

\item {} 
\sphinxAtStartPar
Add fix for bright image canvas on Safari

\item {} 
\sphinxAtStartPar
Add an option to remove markers’ outlines.

\end{itemize}


\subsection{3.0.8.3}
\label{\detokenize{docs/intro/versions:id3}}\begin{itemize}
\item {} 
\sphinxAtStartPar
Fix png artifact in Firefox, by generating jpg tiles.

\end{itemize}


\subsection{3.0.8.2}
\label{\detokenize{docs/intro/versions:id4}}\begin{itemize}
\item {} 
\sphinxAtStartPar
Add high resolution capture of viewport, up to 4096x4096 pixels.

\end{itemize}


\subsection{3.0.8.1}
\label{\detokenize{docs/intro/versions:id5}}\begin{itemize}
\item {} 
\sphinxAtStartPar
Fix multiple dataset alignment when no background image

\end{itemize}


\subsection{3.0.8}
\label{\detokenize{docs/intro/versions:id6}}\begin{itemize}
\item {} 
\sphinxAtStartPar
Fix black images generated by VIPS

\item {} 
\sphinxAtStartPar
Fix Linux and Mac open of captures

\item {} 
\sphinxAtStartPar
Auto save datasets as buttons when saving tmap projects

\item {} 
\sphinxAtStartPar
Add \sphinxcode{\sphinxupquote{mpp}} (microns per pixel) option in tmap files, to add scale bar to viewer

\item {} 
\sphinxAtStartPar
Make region line thickness depend on zoom level

\item {} 
\sphinxAtStartPar
Add compatibility with JupyterLab

\item {} 
\sphinxAtStartPar
Add opacity per marker option

\end{itemize}


\subsection{3.0.7}
\label{\detokenize{docs/intro/versions:id7}}\begin{itemize}
\item {} 
\sphinxAtStartPar
Add menu to load plugins through an update\sphinxhyphen{}site

\end{itemize}


\subsection{3.0.6}
\label{\detokenize{docs/intro/versions:id8}}\begin{itemize}
\item {} 
\sphinxAtStartPar
Fix multiple plugins opening always last plugin

\item {} 
\sphinxAtStartPar
Move to OpenSeadragon 3.0.0

\item {} 
\sphinxAtStartPar
Add tooltip format in Advanced Settings

\item {} 
\sphinxAtStartPar
Add drag and drop to open CSV files and images

\item {} 
\sphinxAtStartPar
Add “Add layer” button for flask version

\item {} 
\sphinxAtStartPar
Add viewport capture

\end{itemize}


\subsection{3.0.5}
\label{\detokenize{docs/intro/versions:id9}}\begin{itemize}
\item {} 
\sphinxAtStartPar
Move csv loading to Papa Parse streaming, to allow better memory management

\end{itemize}


\subsection{3.0.4}
\label{\detokenize{docs/intro/versions:id10}}\begin{itemize}
\item {} 
\sphinxAtStartPar
Add filtering of markers

\end{itemize}


\subsection{3.0}
\label{\detokenize{docs/intro/versions:id11}}\begin{itemize}
\item {} 
\sphinxAtStartPar
Add tissuumaps.jupyter module

\end{itemize}

\sphinxstepscope


\chapter{Getting started}
\label{\detokenize{docs/starting/index:getting-started}}\label{\detokenize{docs/starting/index::doc}}
\sphinxstepscope


\section{Images}
\label{\detokenize{docs/starting/images:images}}\label{\detokenize{docs/starting/images::doc}}

\subsection{Supported image formats}
\label{\detokenize{docs/starting/images:supported-image-formats}}
\sphinxAtStartPar
TissUUmaps can read whole slide images in any format recognized by the OpenSlide library:
\begin{itemize}
\item {} 
\sphinxAtStartPar
Aperio (.svs, .tif)

\item {} 
\sphinxAtStartPar
Hamamatsu (.ndpi, .vms, .vmu)

\item {} 
\sphinxAtStartPar
Leica (.scn)

\item {} 
\sphinxAtStartPar
MIRAX (.mrxs)

\item {} 
\sphinxAtStartPar
Philips (.tiff)

\item {} 
\sphinxAtStartPar
Sakura (.svslide)

\item {} 
\sphinxAtStartPar
Trestle (.tif)

\item {} 
\sphinxAtStartPar
Ventana (.bif, .tif)

\item {} 
\sphinxAtStartPar
Generic tiled TIFF (.tif)

\end{itemize}

\sphinxAtStartPar
TissUUmaps will automatically convert any other format into a pyramidal tiff (in a temporary \sphinxcode{\sphinxupquote{.tissuumaps}} folder created in the original image folder) using vips.

\sphinxAtStartPar
If your image fails to open, try converting it to \sphinxcode{\sphinxupquote{tif}} format using an external tool.


\subsection{Load images}
\label{\detokenize{docs/starting/images:load-images}}



\subsection{Apply filters}
\label{\detokenize{docs/starting/images:apply-filters}}
\sphinxstepscope


\section{Markers}
\label{\detokenize{docs/starting/markers:markers}}\label{\detokenize{docs/starting/markers::doc}}

\subsection{Supported marker format}
\label{\detokenize{docs/starting/markers:supported-marker-format}}
\sphinxAtStartPar
TissUUmaps can read CSV (Comma Separated Values) files with a header row, and at least spatial coordinate columns (X and Y). CSV files are not limited in the number of columns or number of rows. Other columns can contain information for displaying markers (key to group markers, color, size, shape, piecharts, etc.)

\sphinxAtStartPar
CSV files can be exported from any spreadsheet program, or any programming language (Python, R, etc.)


\subsection{Load markers}
\label{\detokenize{docs/starting/markers:load-markers}}

\subsection{Markers settings}
\label{\detokenize{docs/starting/markers:markers-settings}}

\subsubsection{File and coordinates}
\label{\detokenize{docs/starting/markers:file-and-coordinates}}

\subsubsection{Render options}
\label{\detokenize{docs/starting/markers:render-options}}

\subsubsection{Advanced options}
\label{\detokenize{docs/starting/markers:advanced-options}}

\subsubsection{Table of markers}
\label{\detokenize{docs/starting/markers:table-of-markers}}
\sphinxstepscope


\section{Regions}
\label{\detokenize{docs/starting/regions:regions}}\label{\detokenize{docs/starting/regions::doc}}

\subsection{Supported region formats}
\label{\detokenize{docs/starting/regions:supported-region-formats}}
\sphinxAtStartPar
TissUUmaps can read and write region files in the \sphinxhref{https://geojson.org/}{GeoJSON} format.

\sphinxAtStartPar
Only a subset of the GeoJSON format is supported, as TissUUmaps uses only polygonal regions:

\sphinxAtStartPar
\sphinxstylestrong{Main types}:
\begin{itemize}
\item {} 
\sphinxAtStartPar
Feature

\item {} 
\sphinxAtStartPar
FeatureCollection

\item {} 
\sphinxAtStartPar
GeometryCollection

\end{itemize}

\sphinxAtStartPar
\sphinxstylestrong{Geometries}:
\begin{itemize}
\item {} 
\sphinxAtStartPar
Polygon

\item {} 
\sphinxAtStartPar
Multipolygon

\end{itemize}

\sphinxAtStartPar
The coordinate system must be the same as the image and marker coordinate systems.


\subsection{Draw Regions}
\label{\detokenize{docs/starting/regions:draw-regions}}

\subsection{Analyze Regions}
\label{\detokenize{docs/starting/regions:analyze-regions}}

\subsection{Load Regions}
\label{\detokenize{docs/starting/regions:load-regions}}

\subsection{Export Regions}
\label{\detokenize{docs/starting/regions:export-regions}}
\sphinxstepscope


\section{Projects}
\label{\detokenize{docs/starting/projects:projects}}\label{\detokenize{docs/starting/projects::doc}}

\subsection{Saving and loading projects}
\label{\detokenize{docs/starting/projects:saving-and-loading-projects}}

\subsection{The tmap file format}
\label{\detokenize{docs/starting/projects:the-tmap-file-format}}
\sphinxstepscope


\section{Exporting screenshots}
\label{\detokenize{docs/starting/capture:exporting-screenshots}}\label{\detokenize{docs/starting/capture::doc}}


\sphinxstepscope


\section{Plugins}
\label{\detokenize{docs/starting/plugins:plugins}}\label{\detokenize{docs/starting/plugins::doc}}

\subsection{Load plugins}
\label{\detokenize{docs/starting/plugins:load-plugins}}

\subsection{Make your own plugin}
\label{\detokenize{docs/starting/plugins:make-your-own-plugin}}
\sphinxAtStartPar
Download the Plugin Template python and javascript files from the \sphinxhref{https://tissuumaps.github.io/TissUUmaps/plugins/}{Plugin Update Site} and put them in your local folder \sphinxcode{\sphinxupquote{\$USER\_PATH/.tissuumaps/plugins/}}.


\subsubsection{Javascript file}
\label{\detokenize{docs/starting/plugins:javascript-file}}
\sphinxAtStartPar
You can access the javascript API \sphinxhref{https://tissuumaps.github.io/TissUUmapsCore/}{here}.


\subsubsection{Python file}
\label{\detokenize{docs/starting/plugins:python-file}}
\sphinxAtStartPar
You only need to use the Python file if your plugin needs to do processing on the server side. For pure javascript plugins, you can leave this file empty.

\sphinxstepscope


\chapter{Sharing projects}
\label{\detokenize{docs/sharing/index:sharing-projects}}\label{\detokenize{docs/sharing/index::doc}}
\sphinxstepscope


\section{Apache server}
\label{\detokenize{docs/sharing/apache:apache-server}}\label{\detokenize{docs/sharing/apache::doc}}
\sphinxAtStartPar
TissUUmaps projects can be exported into static webpages, that can be uploaded to any Apache server.
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Save your project from TissUUmaps (\sphinxcode{\sphinxupquote{menu \textgreater{} File \textgreater{} Save project}})

\item {} 
\sphinxAtStartPar
Export to static page (\sphinxcode{\sphinxupquote{menu \textgreater{} File \textgreater{} Export to static webpage}})

\item {} 
\sphinxAtStartPar
Copy the exported folder on your Apache server

\end{enumerate}

\sphinxstepscope


\section{Docker container}
\label{\detokenize{docs/sharing/docker:docker-container}}\label{\detokenize{docs/sharing/docker::doc}}\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Start the docker container \sphinxcode{\sphinxupquote{cavenel/tissuumaps:latest}} from Docker Hub:

\end{enumerate}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
docker run \PYGZhy{}it \PYGZhy{}p \PYG{l+m}{56733}:80 \PYGZhy{}\PYGZhy{}name\PYG{o}{=}tissuumaps \PYGZhy{}v /path/to/local/images:/mnt/data cavenel/tissuumaps:latest
\end{sphinxVerbatim}
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Place your images in the local folder \sphinxcode{\sphinxupquote{/path/to/local/images/share}}.

\item {} 
\sphinxAtStartPar
Open \sphinxurl{http://127.0.0.1:56733/} in your favorite browser.

\end{enumerate}

\sphinxstepscope


\chapter{Advanced usage}
\label{\detokenize{docs/advanced/index:advanced-usage}}\label{\detokenize{docs/advanced/index::doc}}
\sphinxstepscope


\section{Jupyter notebooks}
\label{\detokenize{docs/advanced/jupyter:jupyter-notebooks}}\label{\detokenize{docs/advanced/jupyter::doc}}
\sphinxAtStartPar
TissUUmaps can easily be used inside a Jupyter Notebook or Jupyter Lab.

\sphinxAtStartPar
Simple example to load an image in TissUUmaps:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{k+kn}{import} \PYG{n+nn}{tissuumaps}\PYG{n+nn}{.}\PYG{n+nn}{jupyter} \PYG{k}{as} \PYG{n+nn}{tj}
\PYG{n}{viewer} \PYG{o}{=} \PYG{n}{tj}\PYG{o}{.}\PYG{n}{loaddata}\PYG{p}{(}\PYG{p}{[}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{image.png}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{]}\PYG{p}{)}

\PYG{n}{viewer}\PYG{o}{.}\PYG{n}{screenshot}\PYG{p}{(}\PYG{p}{)}
\end{sphinxVerbatim}
\phantomsection\label{\detokenize{docs/advanced/jupyter:module-tissuumaps.jupyter}}\index{module@\spxentry{module}!tissuumaps.jupyter@\spxentry{tissuumaps.jupyter}}\index{tissuumaps.jupyter@\spxentry{tissuumaps.jupyter}!module@\spxentry{module}}

\subsection{tissuumaps.jupyter}
\label{\detokenize{docs/advanced/jupyter:tissuumaps-jupyter}}
\sphinxAtStartPar
Module used to run TissUUmaps from a Jupyter Notebook or from Jupyter Lab.
\index{opentmap() (in module tissuumaps.jupyter)@\spxentry{opentmap()}\spxextra{in module tissuumaps.jupyter}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{docs/advanced/jupyter:tissuumaps.jupyter.opentmap}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{tissuumaps.jupyter.}}\sphinxbfcode{\sphinxupquote{opentmap}}}{\emph{\DUrole{n}{path}}, \emph{\DUrole{n}{port}\DUrole{o}{=}\DUrole{default_value}{5100}}, \emph{\DUrole{n}{host}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}localhost\textquotesingle{}}}, \emph{\DUrole{n}{height}\DUrole{o}{=}\DUrole{default_value}{700}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Open a tmap project
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{path}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The path to a tmap file

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{port}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} The port to run the TissUUmaps server

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{host}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The host to run the TissUUmaps server

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{height}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} The height of the jupyter iframe

\end{itemize}

\item[{Returns}] \leavevmode
\sphinxAtStartPar
The TissUUmaps viewer

\item[{Return type}] \leavevmode
\sphinxAtStartPar
{\hyperref[\detokenize{docs/advanced/jupyter:tissuumaps.jupyter.TissUUmapsViewer}]{\sphinxcrossref{TissUUmapsViewer}}}

\end{description}\end{quote}

\end{fulllineitems}

\index{loaddata() (in module tissuumaps.jupyter)@\spxentry{loaddata()}\spxextra{in module tissuumaps.jupyter}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{docs/advanced/jupyter:tissuumaps.jupyter.loaddata}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxcode{\sphinxupquote{tissuumaps.jupyter.}}\sphinxbfcode{\sphinxupquote{loaddata}}}{\emph{\DUrole{n}{images}\DUrole{o}{=}\DUrole{default_value}{{[}{]}}}, \emph{\DUrole{n}{csvFiles}\DUrole{o}{=}\DUrole{default_value}{{[}{]}}}, \emph{\DUrole{n}{xSelector}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}x\textquotesingle{}}}, \emph{\DUrole{n}{ySelector}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}y\textquotesingle{}}}, \emph{\DUrole{n}{keySelector}\DUrole{o}{=}\DUrole{default_value}{None}}, \emph{\DUrole{n}{nameSelector}\DUrole{o}{=}\DUrole{default_value}{None}}, \emph{\DUrole{n}{colorSelector}\DUrole{o}{=}\DUrole{default_value}{None}}, \emph{\DUrole{n}{piechartSelector}\DUrole{o}{=}\DUrole{default_value}{None}}, \emph{\DUrole{n}{shapeSelector}\DUrole{o}{=}\DUrole{default_value}{None}}, \emph{\DUrole{n}{scaleSelector}\DUrole{o}{=}\DUrole{default_value}{None}}, \emph{\DUrole{n}{fixedShape}\DUrole{o}{=}\DUrole{default_value}{None}}, \emph{\DUrole{n}{scaleFactor}\DUrole{o}{=}\DUrole{default_value}{1}}, \emph{\DUrole{n}{colormap}\DUrole{o}{=}\DUrole{default_value}{None}}, \emph{\DUrole{n}{compositeMode}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}source\sphinxhyphen{}over\textquotesingle{}}}, \emph{\DUrole{n}{boundingBox}\DUrole{o}{=}\DUrole{default_value}{None}}, \emph{\DUrole{n}{port}\DUrole{o}{=}\DUrole{default_value}{5100}}, \emph{\DUrole{n}{host}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}localhost\textquotesingle{}}}, \emph{\DUrole{n}{height}\DUrole{o}{=}\DUrole{default_value}{700}}, \emph{\DUrole{n}{tmapFilename}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}\_project\textquotesingle{}}}, \emph{\DUrole{n}{plugins}\DUrole{o}{=}\DUrole{default_value}{{[}{]}}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Load data in TissUUmaps
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{images}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}\sphinxstyleliteralemphasis{\sphinxupquote{ | }}\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} List of images or single image to display

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{csvFiles}} (list {\color{red}\bfseries{}|}str) \textendash{} List of csv files or single csv file to display

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{xSelector}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the csv column defining the X coordinates

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{ySelector}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the csv column defining the Y coordinates

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{keySelector}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the csv column defining the grouping key

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{nameSelector}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the csv column defining the group name

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{colorSelector}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the csv column defining the group color

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{piechartSelector}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the csv column defining pie\sphinxhyphen{}charts

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{shapeSelector}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the csv column defining markers’ shape

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{scaleSelector}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the csv column defining markers’ scale

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{fixedShape}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} Name of the markers’ shape

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{scaleFactor}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} Global scale of markers

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{colormap}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the colormap used if colorSelector is set

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{compositeMode}} \textendash{} (str): Composite mode used for images

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{boundingBox}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} {[}X,Y,W,H{]} of the bounding box to display

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{port}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} The port to run the TissUUmaps server

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{host}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The host to run the TissUUmaps server

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{height}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}) \textendash{} The height of the jupyter iframe

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{tmapFilename}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the project file that will be created

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{plugins}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List of plugins to add to the tmap project

\end{itemize}

\item[{Returns}] \leavevmode
\sphinxAtStartPar
The TissUUmaps viewer

\item[{Return type}] \leavevmode
\sphinxAtStartPar
{\hyperref[\detokenize{docs/advanced/jupyter:tissuumaps.jupyter.TissUUmapsViewer}]{\sphinxcrossref{TissUUmapsViewer}}}

\end{description}\end{quote}

\end{fulllineitems}

\index{TissUUmapsViewer (class in tissuumaps.jupyter)@\spxentry{TissUUmapsViewer}\spxextra{class in tissuumaps.jupyter}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{docs/advanced/jupyter:tissuumaps.jupyter.TissUUmapsViewer}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{tissuumaps.jupyter.}}\sphinxbfcode{\sphinxupquote{TissUUmapsViewer}}}{\emph{\DUrole{n}{server}}, \emph{\DUrole{n}{image}}, \emph{\DUrole{n}{height}\DUrole{o}{=}\DUrole{default_value}{700}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Class representing a TissUUmaps viewer instance
\index{screenshot() (tissuumaps.jupyter.TissUUmapsViewer method)@\spxentry{screenshot()}\spxextra{tissuumaps.jupyter.TissUUmapsViewer method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{docs/advanced/jupyter:tissuumaps.jupyter.TissUUmapsViewer.screenshot}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{screenshot}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Capture the TissUUmaps viewport and display image in the Notebook.

\end{fulllineitems}


\end{fulllineitems}

\index{TissUUmapsServer (class in tissuumaps.jupyter)@\spxentry{TissUUmapsServer}\spxextra{class in tissuumaps.jupyter}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{docs/advanced/jupyter:tissuumaps.jupyter.TissUUmapsServer}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{  }}}\sphinxcode{\sphinxupquote{tissuumaps.jupyter.}}\sphinxbfcode{\sphinxupquote{TissUUmapsServer}}}{\emph{\DUrole{n}{slideDir}}, \emph{\DUrole{n}{port}\DUrole{o}{=}\DUrole{default_value}{5000}}, \emph{\DUrole{n}{host}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}0.0.0.0\textquotesingle{}}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Class representing a TissUUmaps server instance

\end{fulllineitems}


\sphinxstepscope


\section{Napari}
\label{\detokenize{docs/advanced/napari:napari}}\label{\detokenize{docs/advanced/napari::doc}}
\sphinxAtStartPar
Napari features an important hub containing 118 plugins at the time of writing, many of them expanding further the capabilities of Napari when dealing with biomedical imaging. We thus created our own plugin to allow users to work in Napari, benefit from the tools, scripting and existing plugins, and easily visualize and share the output of their research through TissUUmaps.

\sphinxAtStartPar
The \sphinxhref{https://github.com/TissUUmaps/napari-tissuumaps}{Napari\sphinxhyphen{}TissUUmaps plugin} is available on Napari Hub which makes the installation trivial: from the Napari install/uninstall plugins menu, the \sphinxcode{\sphinxupquote{napari\sphinxhyphen{}tissuumaps}} appears in the list and can be installed with a single click. Alternatively, the plugin can be installed with the Python package manager: \sphinxcode{\sphinxupquote{pip install napari\sphinxhyphen{}tissuumaps}}.

\sphinxAtStartPar
The plugin can export all standard Napari layers, such as images, labels, points, and shapes and preserves the metadata (opacity, visibility), but also the objects parameters (e.g.: label colors, marker colors and symbols, etc…). To export a TissUUmaps project, care must be taken to save all layers of interest and type in a name with the extension \sphinxcode{\sphinxupquote{.tmap}}, e.g.: \sphinxcode{\sphinxupquote{myProject.tmap}}. This is important for Napari to delegate the saving of the files to the plugin. A folder is created and contains all the necessary files and can be loaded in the TissUUmaps server, software, Jupyter Notebook, or shared with the community.

\sphinxAtStartPar
The project folders generated by the plugin contain the metadata in a \sphinxcode{\sphinxupquote{main.tmap}} file, along with folders for each Napari layer types: images, labels, points and regions. Images and labels are saved as plain tif images, points are saved as CSV files, and shapes are saved as GeoJSON. We hope that the use of a simple structure and widespread file formats can simplify the modifying and updating of the TissUUmaps project when prototyping with e.g. Jupyter Notebooks.
The source code is available at \sphinxurl{https://github.com/TissUUmaps/napari-tissuumaps} under the permissive MIT license.
A demonstration of the Cellpose plugin of Napari being exported to the TissUUmaps web viewer is available at: \sphinxurl{https://tissuumaps.github.io/tutorials/\#napari}.

\sphinxstepscope


\section{AnnData}
\label{\detokenize{docs/advanced/anndata:anndata}}\label{\detokenize{docs/advanced/anndata::doc}}
\sphinxAtStartPar
Work in progress



\renewcommand{\indexname}{Index}
\printindex
\end{document}